dotnet/targets/Xamarin.Shared.Sdk.props

```diff
@@ -150,13 +150,13 @@
 	</PropertyGroup>
 	<PropertyGroup Condition="'$(TrimMode)' == ''">
 		<!-- Linking is always on for all assemblies when using NativeAOT - this is because we need to modify all assemblies in the linker for them to be compatible with NativeAOT -->
-		<_DefaultLinkMode Condition="'$(_UseNativeAot)' == 'true'">Full</_DefaultLinkMode>
+		<_DefaultLinkMode Condition="'$(PublishAot)' == 'true'">Full</_DefaultLinkMode>
```

Comment:
This will slow down debug/simulator builds significantly.

How does this work in desktop/console apps? Will setting PublishAot=true enable the trimmer for debug builds as well?

Reply:
We wanted to align projects with `<PublishAot>true</PublishAot>` with projects that have `<TrimMode>full</TrimMode>`. Now that you pointed this out, I'm realizing that we're telling devs to only enable TrimMode=full in Release builds and not in Debug builds and this change is incorrect.

The goal of this PR is to make sure we don't suppress trimming warnings in Debug builds so I'll make sure to achieve that without changing the trim mode.

/cc @ivanpovazan

Reply:
> How does this work in desktop/console apps? Will setting PublishAot=true enable the trimmer for debug builds as well?

Yes, when we set `PublishAot=true` in a project file and do:
```
ivanpovazan@EONE-MAC-ARM64 ~/tmp/abcd $ dotnet build -c Debug -p:PublishAot=true
Restore complete (0.7s)
You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
  abcd succeeded with 1 warning(s) (0.4s) → bin/Debug/net9.0/abcd.dll
    /Users/ivan/tmp/abcd/Program.cs(9,21): warning IL2075: 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicProperties' in call to 'System.Type.GetProperties()'. The return value of method 'System.Object.GetType()' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.

Build succeeded with 1 warning(s) in 1.3s

Workload updates are available. Run `dotnet workload list` for more information.
ivanpovazan@EONE-MAC-ARM64 ~/tmp/abcd $ rm -rf bin obj                          
ivanpovazan@EONE-MAC-ARM64 ~/tmp/abcd $ dotnet build -c Debug -p:PublishAot=false
Restore complete (0.5s)
You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
  abcd succeeded (0.4s) → bin/Debug/net9.0/abcd.dll

Build succeeded in 1.1s
```